internal/sdkv2provider/resource_user_identity_test.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package sdkv2
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func testAccResourceUserIdentity(t *testing.T) resource.TestCase {
+	return resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: configResourceBasicIdentity,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr(
+						"corner_user_identity.foo", "name", regexp.MustCompile("^For")),
+				),
+				PlanOnly:           true, // Should have a plan with something in there and not error
+				ExpectNonEmptyPlan: true,
```

Comment:
Interesting, I'd think we'd want to run a full plan/apply here, so it actually executes the CRUD operation (perhaps also have a second test step to test the update?). With this setup, it won't create the resource at all, so we won't test read (no refresh since it doesn't exist) or create.

Is there a specific reason you added these?

Reply:
nope, it was mostly in the process of trying to get the test going until I noticed, that I can't actually check the contents of the id via the test framework yet 😄

Reply:
I'll improve the test 👍

Reply:
> nope, it was mostly in the process of trying to get the test going until I noticed, that I can't actually check the contents of the id via the test framework yet 😄

Right! I was going to take that task on next actually, should be a relatively quick addition. If you want to just make sure the test steps apply (basically just no checks) then I think that's good for an MVP to merge this 👍🏻 

I can come back through and update this test in a follow-up PR when the `terraform-plugin-testing` logic is available.

Reply:
pushed a commit – I had originally searched for some `Apply` property for the test step but couldn't find one 😅 Should've read the docs 😂

Reply:
Yeah, by default the test step will run:

`terraform plan`
`terraform apply`
`terraform plan`

And possibly another plan? I can't remember 😆